agent/stats/queue_test.go

```diff
@@ -257,6 +339,26 @@ func TestQueueAddPredictableHighMemoryUtilization(t *testing.T) {
 	}
 }
 
+func TestUintOverflow(t *testing.T) {
```

Comment:
What are your thoughts on rewriting this as a table-driven test?

Another thing I'd like to see here is uint64(math.MaxInt64) + 1

Reply:
In my eyes, the `baseOverUint, overflowOverUint := getInt64WithOverflow(overUint)` check is checking the same thing as `uint64(math.MaxInt64) + 1`

`overUint = uint64(maxInt) + uint64(overInt)` is essentially `uint64(math.MaxInt64) + 20`
Would you prefer I change the 20 to a 1?

Reply:
and as for the table, I think that only 3 checks are necessary: `<`, `=` and `>` so it doesn't really need to get fancier.  I could narrow the `<` and `>` range to 1 if you think it would make a difference.

Reply:
The QA in me likes thinking about "arbitrarily larger" and "maxint + 1" as different cases, since "maxint+1" is where the overflow would actually occur.

Reply:
made this update.

Reply:
^ nvm, we talked offline and what you have makes sense.

n-1 and n cover the overflow case. (The plot twist is that *I* was the off by one error the whole time)